src/components/Menus/DropdownMenu.js

```diff
@@ -0,0 +1,207 @@
+import React             from 'react'
+import PropTypes         from 'prop-types'
+import Radium            from 'radium'
+import Menu              from './Menu'
+
+const styles = {
+  menuContainer: {
+    zIndex: 9999,
+    position: 'absolute',
+    display: 'none',
+    width: '100%'
+  },
+  menuContainerOpen: {
+    display: 'block'
+  }
+}
+
+@Radium
+class DropdownMenu extends React.Component {
+  static propTypes = {
+    /** MenuItems or Divider */
+    children: PropTypes.node.isRequired,
+    /** React component or HTML that will be used as the dropdown trigger. ie. Button or Icon */
+    triggerElement: PropTypes.element,
+    /** Open and close menu manually */
+    open: PropTypes.bool,
+    /** Callback fired on dropdown close */
+    onClose: PropTypes.func,
+    /** Callback fired on click */
+    onClick: PropTypes.func,
+    /** Callback fired on dropdown open */
+    onOpen: PropTypes.func,
+    /** Callback function fired on key down */
+    onKeyDown: PropTypes.func,
+    /** Requesting to open/close for controlled open component. */
+    onRequestChange: PropTypes.func,
+    /** Callback function fired when a MenuItem is selected */
+    onSelect: PropTypes.func,
+    /** Customize style of menu parent */
+    style: PropTypes.shape({}),
+    /** Props passed to Menu component */
+    menuProps: PropTypes.shape({})
+  }
+
+  static defaultProps = {
+    open: null
+  }
+
+  state = {
+    open: false
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.open !== null && this.props.open !== nextProps.open) {
+      if (nextProps.open) {
+        this.open()
+      } else {
+        this.close()
+      }
+    }
+  }
+
+  _handleSelect = (option) => {
+    const { onSelect, onRequestChange } = this.props
+
+    if (this.controlledOpen()) {
+      onRequestChange && onRequestChange(false)
+    } else {
+      // if multi then do not close
+      this.close()
+    }
+
+    onSelect && onSelect(option)
+  }
+
+  _handleClick = (event) => {
+    event.preventDefault()
+    const { open, onClick, onRequestChange } = this.props
+
+    if (this.controlledOpen()) {
+      onRequestChange && onRequestChange(!open)
+      return
+    }
+
+    if (!this.state.open) {
+      this.open()
+    } else {
+      this.close()
+    }
+    onClick && onClick(event)
+  }
+
+  _handleMenuBlur = () => {
+    const { onRequestChange } = this.props
+    // Timeout ensures click event has correct state when attempting to close
```

Comment:
What is the event you are worried about here?

Reply:
When a MenuItem is selected the Menu is blur

Reply:
Sorry, I still don't understand the issue - the comment above says that the timeout ensures the click event will have the correct state. What is the click event and what is the correct state that is not correct without the timeout?